ddtrace/tracer/tracer.go

```diff
@@ -286,7 +289,7 @@ func (t *tracer) worker(tick <-chan time.Time) {
 		case trace := <-t.out:
 			t.sampleFinishedTrace(trace)
 			if len(trace.spans) != 0 {
-				t.traceWriter.add(trace.spans)
+				t.writeTrace(trace.spans)
```

Comment:
Here, we could run the processor before or after single span sampling.

Running the processor before single span sampling:
- If the trace is dropped, no need to call `sampleFinishedTrace`.
- There could be a use case where the customer want's to add a tag on a span that is kept which is based on some of the spans that aren't kept (e.g. having a tag that indicates the number of db spans).

Running the processor after single span sampling:
- Spans dropped won't run in the processor (no need to modify spans in the processor that will be dropped).
- This is in line with what we do for P0s => drop P0's pre-processor.

Currently, I went with the second option, but am open to changing this.

Reply:
Whichever we decide, let's be sure to document it to set user expectations.
My inclination would to do the latter, i.e. run the processing after single span sampling. I think the logic is clearer that way, and it saves us the bandwidth of running dropped spans in the processor as you said. Let's see what others think too.